mqtt/mqttd/src/app/cleanup.rs

```diff
@@ -0,0 +1,36 @@
+use std::time::Duration as StdDuration;
+
+use anyhow::Result;
+use chrono::{Duration, Utc};
+use tokio::time::{self, Instant};
+use tracing::{info, warn};
+
+use mqtt_broker::{BrokerHandle, Message, SystemEvent};
+
+pub async fn start_cleanup(
+    broker_handle: BrokerHandle,
+    cleanup_interval: StdDuration,
+    expiration: StdDuration,
+) -> Result<()> {
+    info!("starting cleanup job...");
+
+    let expiration = Duration::from_std(expiration)?;
+    let tick = tick_cleanup(cleanup_interval, expiration, broker_handle);
+    tokio::spawn(tick);
```

Comment:
Do we need to handle shutdown?

Reply:
good point. I kept it similar to the Snapshotter tick and skipped the shutdown. We can revisit that later.